test/functional/ws/transport.handshake.js

```diff
@@ -87,63 +88,57 @@ describe('handshake', function () {
 
 		describe('should fail with INVALID_HEADERS code and description', function () {
 
-			it('without headers', function (done) {
-				delete validClientSocketOptions.query;
-				connect();
-				expectDisconnect(this, function (code) {
+			it('without headers', function () {
+				return new WSClient(null, {disconnect: function (code, description) {
 					expect(code).equal(failureCodes.INVALID_HEADERS);
-					done();
-				});
+				}});
 			});
 
-			it('with empty headers', function (done) {
-				validClientSocketOptions.query = {};
-				connect();
-				expectDisconnect(this, function (code, description) {
+			it('with empty headers', function () {
+				return new WSClient({}, {disconnect: function (code, description) {
 					expect(code).equal(failureCodes.INVALID_HEADERS);
 					expect(description).contain('Missing required property');
-					done();
-				});
+				}}).start();
```

Comment:
So in the first part of the test we are using new approach relying on `WSClient` with the handler (disconnect) passed into to verify if the WS connection succeeds or fails. In the `with valid headers` section, we are using existing approach using functions `connect` and then `expectConnect`. 
Please choose one and apply in both sections.

Reply:
Had already spent hour with you. Unless we resolve the socket-hangup issue its not feasible to move those complex test cases. Once we have connection id on client side, all tests will be refactored well. Using a good approach where possible for now, is a good approach. Please suggest.